DriveKitDriverDataUI/DrivingConditions/Components/DrivingConditionsContext/ViewModels/ConditionsRoadContextViewModel.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  ConditionsRoadContextViewModel.swift
+//  DriveKitDriverDataUI
+//
+//  Created by Amine Gahbiche on 14/04/2023.
+//  Copyright © 2023 DriveQuant. All rights reserved.
+//
+
+import UIKit
+import DriveKitCommonUI
+import DriveKitDBTripAccessModule
+
+class ConditionsRoadContextViewModel {
+    private var contextItems: [ConditionsRoadContextItem] = []
+    var viewModelDidUpdate: (() -> Void)?
+
+    func configure(with distanceByRoadContext: [DKRoadContext: Double]) {
+        var tempItems: [ConditionsRoadContextItem] = []
+        if let distance = distanceByRoadContext[.heavyUrbanTraffic], distance > 0 {
+            tempItems.append(.heavyUrbanTraffic(distance: distance))
+        }
+        if let distance = distanceByRoadContext[.city], distance > 0 {
+            tempItems.append(.city(distance: distance))
+        }
+        if let distance = distanceByRoadContext[.suburban], distance > 0 {
+            tempItems.append(.suburban(distance: distance))
+        }
+        if let distance = distanceByRoadContext[.expressways], distance > 0 {
+            tempItems.append(.expressways(distance: distance))
+        }
+        self.contextItems = tempItems
+        self.viewModelDidUpdate?()
+    }
+}
+
+extension ConditionsRoadContextViewModel: DKContextCard {
+    var items: [DriveKitCommonUI.DKContextItem] {
+        return self.contextItems
+    }
+    
+    var title: String {
+        var cityDistance: Double = 0
+        var expressWayDistance: Double = 0
+        var suburbanDistance: Double = 0
+        for item in contextItems {
+            switch item {
+                case .heavyUrbanTraffic(let distance):
+                    cityDistance += distance
+                case .city(let distance):
+                    cityDistance += distance
```

Comment:
Je pense qu'en mettant ces deux cases ensembles, on met mieux en avant l'intention qui est derrière, c'est à dire le fait qu'on considère les deux cas comme de la "ville"

```suggestion
                case .heavyUrbanTraffic(let distance),
                     .city(let distance):
                    cityDistance += distance
```